model/src/main/proto/preprocess_protos.bzl

```diff
@@ -0,0 +1,21 @@
+
+def process(name, src):
+  """
+  Formats the import statement of the src file to contain a full path to the file
+  The genrule() rule generates a copy of the src file and alters the copy
+  The proto_library() rule takes this altered copy and builds it
+  """
+  native.genrule(
+    name = name,
+    srcs = [src],
+    outs = ["processed_" + src],
+    cmd = ''' cp $< $@ &&\
+    sed 's/import \"/import \"model\/src\/main\/proto\//g' $< > $@
```

Comment:
Out of curiosity, will this work on Windows systems?

Reply:
Good catch. Just read into it and I believe I can add an attribute like `cmd_bat` or `cmd_ps` that specifically runs on Windows (using Batch or Powershell respectively). I am not sure though how I would test if the commands are working though?

Reply:
Suggest filing an issue to verify windows support at the end of phase 1. I have a windows machine that I can test on, but I think it'd be better to make sure that the whole pipeline works on a predictable system (Unix-based) before delving into potential windows-specific issues.

Reply:
Sounds good!

Reply:
#1543 for reference & to link it to this PR.